NuRadioReco/modules/io/coreas/readCoREASStation.py

```diff
@@ -45,8 +46,23 @@ def run(self, detector):
             self.__current_event = 0
             with h5py.File(input_file, "r") as corsika:
                 if "highlevel" not in corsika.keys() or list(corsika["highlevel"].values()) == []:
-                    logger.warning(" No highlevel quantities in simulated hdf5 files, weights will be one")
-                    weights = np.ones(len(corsika['CoREAS']['observers']))
+                    logger.warning(" No highlevel quantities in simulated hdf5 files, weights will be taken from station position")
+                    positions = []
+                    for observer in corsika['CoREAS']['observers'].values():
+                        position = observer.attrs['position']
+                        positions.append(np.array([-position[1], position[0], 0]) * units.cm)
+                    positions = np.array(positions)
+                    zenith, azimuth, magnetic_field_vector = coreas.get_angles(corsika)
+                    weights = coreas.calculate_simulation_weights(positions, zenith, azimuth)
+
+                    # import matplotlib.pyplot as plt
+
+                    # fig, ax = plt.subplots()
+                    # im = ax.scatter(positions[:,0], positions[:,1], c=weights)
+                    # fig.colorbar(im, ax=ax)
+                    # plt.gca().set_aspect('equal')
+                    # plt.show()
+
                 else:
                     weights = coreas.calculate_simulation_weights(list(corsika["highlevel"].values())[0]["antenna_position"])
```

Comment:
this line will crash as the function expects a zenith and azimuth attribute

Reply:
ah sure, i will change it,